### PR TITLE
Tests for MLPack support: dbscan, kmeans

### PR DESCRIPTION
**Pull request by pearu**
Requires https://github.com/omnisci/omniscidb-internal/pull/5430

### CODE REVIEW

**Comment by guilhermeleobas** [comment]
I got the following output when using pytest:

```bash
~/git/rbc:pearu/mlpack-tests (rbc-dev) guilhermel@qgpu3 $ pytest rbc/tests/test_omnisci_mlpack.py -rs -sv
========================================================================================================== test session starts ==========================================================================================================
platform linux -- Python 3.8.6, pytest-6.2.2, py-1.10.0, pluggy-0.13.1 -- /home/guilhermel/.conda/envs/rbc-dev/bin/python
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/home/guilhermel/git/rbc/.hypothesis/examples')
rootdir: /home/guilhermel/git/rbc, configfile: setup.cfg
plugins: hypothesis-6.4.0
collected 2 items

rbc/tests/test_omnisci_mlpack.py::test_mlpack[db_scan]  OmnisciDB version (5, 7, 0, 'dev-20210406-dd992f39d9')
SKIPPED (test requires omniscidb server with MLPACK support: SQL Error: From line 2, column 12 to line 3, column 53: No match found for function signature db_scan(<CU...)
rbc/tests/test_omnisci_mlpack.py::test_mlpack[kmeans] PASSED

======================================================================================================== short test summary info ========================================================================================================
SKIPPED [1] rbc/tests/test_omnisci_mlpack.py:32: test requires omniscidb server with MLPACK support: SQL Error: From line 2, column 12 to line 3, column 53: No match found for function signature db_scan(<CURSOR>, <NUMERIC>, <NUMERIC>, <NUMERIC>)
===================================================================================================== 1 passed, 1 skipped in 0.96s ======================================================================================================
```

Is this the expected output?

**Comment by pearu** [comment]
> I got the following output when using pytest:
...
> Is this the expected output? `kmeans` worked just fine but `db_scan` query failed.

If omniscidb is built with mlpack support, then no.

File: rbc/tests/test_omnisci_mlpack.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+
+@pytest.fixture(scope='module')
+def omnisci():
+
+    for o in omnisci_fixture(globals(), minimal_version=(5, 6)):
+        yield o
+
+
+@pytest.mark.parametrize("func", ['db_scan', 'kmeans'])
```
Comment (guilhermeleobas) [inline rbc/tests/test_omnisci_mlpack.py:14]: The name is actually `dbscan`, without the underscore.

```suggestion
@pytest.mark.parametrize("func", ['dbscan', 'kmeans'])
```

File: rbc/tests/test_omnisci_mlpack.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+def test_mlpack(omnisci, func):
+    omnisci.require_version(
+        (5, 6),
+        'Requires omniscidb-internal PR 5430 and omniscidb built with -DENABLE_MLPACK',
+        date=20210401)
+
+    extra_args = dict(db_scan='cast(1 as float), 1',
```
Comment (guilhermeleobas) [inline rbc/tests/test_omnisci_mlpack.py:21]:
```suggestion
    extra_args = dict(dbscan='cast(1 as float), 1',
```

File: rbc/tests/test_omnisci_mlpack.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        m = re.match(r'.*No match found for function signature ' + func + r'[(]',
+                     msg.args[0])
+        if m is not None:
+            pytest.skip(f'test requires omniscidb server with MLPACK support: {msg}')
+        raise
+
+    result = list(result)
+
+    expected = dict(
+        db_scan=[(0, 0), (1, 1), (2, 2), (3, 3), (4, 4)],
```
Comment (guilhermeleobas) [inline rbc/tests/test_omnisci_mlpack.py:38]:
```suggestion
        dbscan=[(0, 0), (1, 1), (2, 2), (3, 3), (4, 4)],
```

**Review by guilhermeleobas — CHANGES_REQUESTED** [review]
LGMT! Just a minor change

**Comment by guilhermeleobas** [comment]
Also, do you think it is a good idea for RBC to know if the server has support for armadillo/mlpack?

**Pull request merged**